lib/avrora/avro_decoder_options.ex

```diff
@@ -10,15 +10,17 @@ defmodule Avrora.AvroDecoderOptions do
     encoding: :avro_binary,
     hook: &__MODULE__.__hook__/4,
     is_wrapped: true,
-    map_type: :proplist,
     record_type: :map
   }
   @null_type_name "null"
 
   @doc """
   A unified erlavro decoder options compatible for both binary and OCF decoders.
   """
-  def options, do: @options
+  def options do
+    @options
+    |> Map.put(:map_type, if(convert_map_to_proplist(), do: :proplist, else: :map))
```

Comment:
I have a suggestion, let's make the "main" path fast and update options only if `:proplist` should be set. In other words `:map` should be set in the `@options` by default and modified if needed.